From a9cfb0bcef6fa2316b54cf548bd896f052e0cf8c Mon Sep 17 00:00:00 2001
From: Jason Smith <jsmith@habana.ai>
Date: Sat, 24 May 2025 10:30:42 -0400
Subject: [PATCH] [SW-230673] pldm: auto-gen the pdr database

Use script from libpldm to auto-generate the PDR database.

Change-Id: I7f60665f4c8e914a90be9c447d984d9b2d3845ea
Signed-off-by: Jason Smith <jsmith@habana.ai>
---

diff --git a/common/include/gaudi3/pldm_pdr_generated.h b/common/include/gaudi3/pldm_pdr_generated.h
index d3dbfa1..17dda31 100644
--- a/common/include/gaudi3/pldm_pdr_generated.h
+++ b/common/include/gaudi3/pldm_pdr_generated.h
@@ -3,10 +3,21 @@
  * Copyright (c) 2024 Habana Labs
  */
 
+/* !!! AUTO-GENERATED FILE - DO NOT EDIT !!! */
+/*
+ * Spreadsheet : '20250530_GAUDI3_RAS_PLDM_7th_stage.xlsx'
+ * MD5         : 7234b8d63e15e174b43db71221cb8587
+ *
+ * Command line:
+ *   ./g3-gen-pdrs.py \
+ *     -w "20250530_GAUDI3_RAS_PLDM_7th_stage.xlsx" \
+ *     -r ".../pldm_pdr.h" "pdrs" \
+ *     --header ".../pldm_pdr_generated.h" \
+ *     --source ".../pldm_pdr_generated.c"
+ */
+
 #ifndef PLUGINS_RAS_PLDM_PDR_GENERATED_H_
 #define PLUGINS_RAS_PLDM_PDR_GENERATED_H_
-
-/* NOTE: temporary hand-crafted file; will be auto-generated in the future */
 
 #include "pldm.h"
 #include <pldm_arch_common.h>
@@ -68,4 +79,3 @@
 extern const size_t pldm_pdr_oem_unit_records_length;
 
 #endif /* PLUGINS_RAS_PLDM_PDR_GENERATED_H_ */
-
diff --git a/common/src/gaudi3/ras/pldm/pldm_pdr_generated.c b/common/src/gaudi3/ras/pldm/pldm_pdr_generated.c
index ef1c074..e6684f9 100644
--- a/common/src/gaudi3/ras/pldm/pldm_pdr_generated.c
+++ b/common/src/gaudi3/ras/pldm/pldm_pdr_generated.c
@@ -3,1240 +3,1805 @@
  * Copyright (c) 2024 Habana Labs
  */
 
-/* NOTE: temporary hand-crafted file; will be auto-generated in the future */
+/* !!! AUTO-GENERATED FILE - DO NOT EDIT !!! */
+/*
+ * Spreadsheet : '20250530_GAUDI3_RAS_PLDM_7th_stage.xlsx'
+ * MD5         : 7234b8d63e15e174b43db71221cb8587
+ *
+ * Command line:
+ *   ./g3-gen-pdrs.py \
+ *     -w "20250530_GAUDI3_RAS_PLDM_7th_stage.xlsx" \
+ *     -r ".../pldm_pdr.h" "pdrs" \
+ *     --header ".../pldm_pdr_generated.h" \
+ *     --source ".../pldm_pdr_generated.c"
+ */
 
-#include <pldm_pdr.h>
-#include <pldm_pdr_generated.h>
+#include "pldm_pdr.h"
+#include "pldm_pdr_generated.h"
 #include <pldm_state_set.h>
 #include <pldm_arch_common.h>
 
-const uint16_t oamAddInCardAccelerator_str[] = {
-	0x6f00, 0x6100, 0x6d00, 0x4100, 0x6400, 0x6400, 0x4900,
-	0x6e00, 0x4300, 0x6100, 0x7200, 0x6400, 0x0000
-};
-const uint16_t oamCPLD_str[] = {
-	0x6f00, 0x6100, 0x6d00, 0x4400, 0x5000, 0x4c00, 0x4400,
-	0x0000
-};
-const uint16_t accelerator_str[] = {
-	0x6100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500, 0x7200,
-	0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
-};
-const uint16_t asicReset_str[] = {
-	0x6100, 0x7300, 0x6900, 0x6300, 0x5200, 0x6500, 0x7300,
-	0x6500, 0x7400, 0x0000,
-};
-const uint16_t bootStage_str[] = {
-	0x6200, 0x6f00, 0x6f00, 0x7400, 0x5300, 0x7400, 0x6100,
-	0x6700, 0x6500, 0x0000
-};
-const uint16_t fwUpdateState_str[] = {
-	0x6600, 0x7700, 0x5500, 0x7000, 0x6400, 0x6100, 0x7400,
-	0x6500, 0x5300, 0x7400, 0x6100, 0x7400, 0x6500, 0x0000
-};
+/**** Auxilary Name PDRs ****/
 
-const uint16_t temperature_str[] = {
-	0x7400, 0x6500, 0x6d00, 0x7000, 0x6500, 0x7200, 0x6100,
-	0x7400, 0x7500, 0x7200, 0x6500, 0x0000
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+
+/* Gaudi3 identification ASIC serial number high */
+#define _asicSerialNumberHigh_str "asicSerialNumberHigh"
+/* Gaudi3 identification ASIC serial number low */
+#define _asicSerialNumberLow_str "asicSerialNumberLow"
+/* Gaudi3 identification Device PDR repository version */
+#define _devicePDRRepositoryVersion_str "devicePDRRepositoryVersion"
+/* Gaudi3 PCIe link width */
+#define _currentPCIeLinkWidth_str "currentPCIeLinkWidth"
+/* Gaudi3 PCIe link speed */
+#define _currentPCIeLinkSpeed_str "currentPCIeLinkSpeed"
+/* Gaudi3 FW state */
+#define _bootStage_str "bootStage"
+#define _fwUpdateState_str "fwUpdateState"
+/* Gaudi3 security keys state */
+#define _key1Revocation_str "key1Revocation"
+#define _key2Revocation_str "key2Revocation"
+#define _key3Revocation_str "key3Revocation"
+#define _key4Revocation_str "key4Revocation"
+#define _key5Revocation_str "key5Revocation"
+/* Gaudi3 security state */
+#define _fwImageSource_str "fwImageSource"
+#define _memoryCleanup_str "memoryCleanup"
+#define _spiWriteProtect_str "spiWriteProtect"
+#define _permanentSecured_str "permanentSecured"
+/* Gaudi3 Error state sensor 2 */
+#define _temperature_str "temperature"
+#define _voltage_str "voltage"
+#define _tempRiseTimeErr_str "tempRiseTimeErr"
+#define _hbmCATTRIP_str "hbmCATTRIP"
+#define _security_str "security"
+#define _wd_str "wd"
+#define _checksum_str "checksum"
+/* Gaudi3 security SPI error state sensor */
+#define _revokedPublicKey_str "revokedPublicKey"
+#define _svn_str "svn"
+#define _signature_str "signature"
+/* Gaudi3 reset effecter */
+#define _asicReset_str "asicReset"
+/* Gaudi3 margin tool effecter */
+#define _marginTool_str "marginTool"
+/* Gaudi3 margin tool SERDES OAM state sensor */
+#define _oamResult1_str "oamResult1"
+#define _oamResult2_str "oamResult2"
+#define _oamResult3_str "oamResult3"
+#define _oamResult4_str "oamResult4"
+#define _oamResult5_str "oamResult5"
+#define _oamResult6_str "oamResult6"
+#define _oamResult7_str "oamResult7"
+#define _oamResult8_str "oamResult8"
+/* Gaudi3 margin tool SERDES retimer state sensor 1 */
+#define _retimer1Result1_str "retimer1Result1"
+#define _retimer1Result2_str "retimer1Result2"
+#define _retimer1Result3_str "retimer1Result3"
+#define _retimer1Result4_str "retimer1Result4"
+#define _retimer1Result5_str "retimer1Result5"
+#define _retimer1Result6_str "retimer1Result6"
+/* Gaudi3 margin tool SERDES retimer state sensor 2 */
+#define _retimer2Result1_str "retimer2Result1"
+#define _retimer2Result2_str "retimer2Result2"
+#define _retimer2Result3_str "retimer2Result3"
+#define _retimer2Result4_str "retimer2Result4"
+/* Gaudi3 margin tool PCIe lanes state effecter 1 */
+#define _pcieMarginToolModePort0_str "pcieMarginToolModePort0"
+#define _pcieMarginToolModePort1_str "pcieMarginToolModePort1"
+#define _pcieMarginToolModePort2_str "pcieMarginToolModePort2"
+#define _pcieMarginToolModePort3_str "pcieMarginToolModePort3"
+#define _pcieMarginToolModePort4_str "pcieMarginToolModePort4"
+#define _pcieMarginToolModePort5_str "pcieMarginToolModePort5"
+#define _pcieMarginToolModePort6_str "pcieMarginToolModePort6"
+#define _pcieMarginToolModePort7_str "pcieMarginToolModePort7"
+/* Gaudi3 margin tool PCIe lanes state effecter 2 */
+#define _pcieMarginToolModePort8_str "pcieMarginToolModePort8"
+#define _pcieMarginToolModePort9_str "pcieMarginToolModePort9"
+#define _pcieMarginToolModePort10_str "pcieMarginToolModePort10"
+#define _pcieMarginToolModePort11_str "pcieMarginToolModePort11"
+#define _pcieMarginToolModePort12_str "pcieMarginToolModePort12"
+#define _pcieMarginToolModePort13_str "pcieMarginToolModePort13"
+#define _pcieMarginToolModePort14_str "pcieMarginToolModePort14"
+#define _pcieMarginToolModePort15_str "pcieMarginToolModePort15"
+/* Gaudi3 identification PCIe device ID */
+#define _pcieDeviceID_str "pcieDeviceID"
+/* Gaudi3 identification PCIe vendor ID */
+#define _pcieVendorID_str "pcieVendorID"
+
+#else /* old aux name format */
+
+/* Gaudi3 identification ASIC serial number high */
+const uint16_t _asicSerialNumberHigh_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6100, 0x7300, 0x6900, 0x6300, 0x5300, 0x6500,
+	0x7200, 0x6900, 0x6100, 0x6c00, 0x4e00, 0x7500,
+	0x6d00, 0x6200, 0x6500, 0x7200, 0x4800, 0x6900,
+	0x6700, 0x6800, 0x0000
 };
-const uint16_t voltage_str[] = {
-	0x7600, 0x6f00, 0x6c00, 0x7400, 0x6100, 0x6700, 0x6500,
-	0x0000
+/* Gaudi3 identification ASIC serial number low */
+const uint16_t _asicSerialNumberLow_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6100, 0x7300, 0x6900, 0x6300, 0x5300, 0x6500,
+	0x7200, 0x6900, 0x6100, 0x6c00, 0x4e00, 0x7500,
+	0x6d00, 0x6200, 0x6500, 0x7200, 0x4c00, 0x6f00,
+	0x7700, 0x0000
 };
-const uint16_t hbmCATTRIP_str[] = {
-	0x6800, 0x6200, 0x6d00, 0x4300, 0x4100, 0x5400, 0x5400,
-	0x5200, 0x4900, 0x5000, 0x0000
+/* Gaudi3 identification Device PDR repository version */
+const uint16_t _devicePDRRepositoryVersion_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6400, 0x6500, 0x7600, 0x6900, 0x6300, 0x6500,
+	0x5000, 0x4400, 0x5200, 0x5200, 0x6500, 0x7000,
+	0x6f00, 0x7300, 0x6900, 0x7400, 0x6f00, 0x7200,
+	0x7900, 0x5600, 0x6500, 0x7200, 0x7300, 0x6900,
+	0x6f00, 0x6e00, 0x0000
 };
-const uint16_t security_str[] = {
-	0x7300, 0x6500, 0x6300, 0x7500, 0x7200, 0x6900, 0x7400,
-	0x7900, 0x0000
+/* Gaudi3 PCIe link width */
+const uint16_t _currentPCIeLinkWidth_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6300, 0x7500, 0x7200, 0x7200, 0x6500, 0x6e00,
+	0x7400, 0x5000, 0x4300, 0x4900, 0x6500, 0x4c00,
+	0x6900, 0x6e00, 0x6b00, 0x5700, 0x6900, 0x6400,
+	0x7400, 0x6800, 0x0000
 };
-const uint16_t wd_str[] = {
-	0x7700, 0x6400, 0x0000
+/* Gaudi3 PCIe link speed */
+const uint16_t _currentPCIeLinkSpeed_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6300, 0x7500, 0x7200, 0x7200, 0x6500, 0x6e00,
+	0x7400, 0x5000, 0x4300, 0x4900, 0x6500, 0x4c00,
+	0x6900, 0x6e00, 0x6b00, 0x5300, 0x7000, 0x6500,
+	0x6500, 0x6400, 0x0000
 };
-const uint16_t checksum_str[] = {
-	0x6300, 0x6800, 0x6500, 0x6300, 0x6b00, 0x7300, 0x7500,
-	0x6d00, 0x0000
+/* Gaudi3 FW state */
+const uint16_t _bootStage_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6200, 0x6f00, 0x6f00, 0x7400, 0x5300, 0x7400,
+	0x6100, 0x6700, 0x6500, 0x0000
 };
-const uint16_t tempRiseTimeErr_str[] = {
-	0x7400, 0x6500, 0x6d00, 0x7000, 0x5200, 0x6900,
-	0x7300, 0x6500, 0x5400, 0x6900, 0x6d00, 0x6500,
-	0x4500, 0x7200, 0x7200, 0x0000
+const uint16_t _fwUpdateState_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6600, 0x7700, 0x5500, 0x7000, 0x6400, 0x6100,
+	0x7400, 0x6500, 0x5300, 0x7400, 0x6100, 0x7400,
+	0x6500, 0x0000
 };
-const uint16_t fwUpdateError_str[] = {
-	0x6600, 0x7700, 0x5500, 0x7000, 0x6400, 0x6100, 0x7400,
-	0x6500, 0x4500, 0x7200, 0x7200, 0x6f00, 0x7200, 0x0000
-};
-const uint16_t revokedPublicKey_str[] = {
-	0x7200, 0x6500, 0x7600, 0x6f00, 0x6b00, 0x6500, 0x6400,
-	0x5000, 0x7500, 0x6200, 0x6c00, 0x6900, 0x6300, 0x4b00,
-	0x6500, 0x7900, 0x0000
-};
-const uint16_t svn_str[] = {
-	0x7300, 0x7600, 0x6e00, 0x0000
-};
-const uint16_t signature_str[] = {
-	0x7300, 0x6900, 0x6700, 0x6e00, 0x6100, 0x7400, 0x7500,
-	0x7200, 0x6500, 0x0000
-};
-const uint16_t key1Revocation_str[] = {
+/* Gaudi3 security keys state */
+const uint16_t _key1Revocation_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x6b00, 0x6500, 0x7900, 0x3100, 0x5200, 0x6500,
 	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
 	0x6f00, 0x6e00, 0x0000
 };
-const uint16_t key2Revocation_str[] = {
+const uint16_t _key2Revocation_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x6b00, 0x6500, 0x7900, 0x3200, 0x5200, 0x6500,
 	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
 	0x6f00, 0x6e00, 0x0000
 };
-const uint16_t key3Revocation_str[] = {
+const uint16_t _key3Revocation_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x6b00, 0x6500, 0x7900, 0x3300, 0x5200, 0x6500,
 	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
 	0x6f00, 0x6e00, 0x0000
 };
-const uint16_t key4Revocation_str[] = {
+const uint16_t _key4Revocation_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x6b00, 0x6500, 0x7900, 0x3400, 0x5200, 0x6500,
 	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
 	0x6f00, 0x6e00, 0x0000
 };
-const uint16_t key5Revocation_str[] = {
+const uint16_t _key5Revocation_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x6b00, 0x6500, 0x7900, 0x3500, 0x5200, 0x6500,
 	0x7600, 0x6f00, 0x6300, 0x6100, 0x7400, 0x6900,
 	0x6f00, 0x6e00, 0x0000
 };
-const uint16_t fwImageSource_str[] = {
+/* Gaudi3 security state */
+const uint16_t _fwImageSource_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x6600, 0x7700, 0x4900, 0x6d00, 0x6100, 0x6700,
 	0x6500, 0x5300, 0x6f00, 0x7500, 0x7200, 0x6300,
 	0x6500, 0x0000
 };
-const uint16_t memoryCleanup_str[] = {
+const uint16_t _memoryCleanup_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x6d00, 0x6500, 0x6d00, 0x6f00, 0x7200, 0x7900,
 	0x4300, 0x6c00, 0x6500, 0x6100, 0x6e00, 0x7500,
 	0x7000, 0x0000
 };
-const uint16_t spiWriteProtect_str[] = {
+const uint16_t _spiWriteProtect_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7300, 0x7000, 0x6900, 0x5700, 0x7200, 0x6900,
 	0x7400, 0x6500, 0x5000, 0x7200, 0x6f00, 0x7400,
 	0x6500, 0x6300, 0x7400, 0x0000
 };
-const uint16_t permanentSecured_str[] = {
+const uint16_t _permanentSecured_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6500, 0x7200, 0x6d00, 0x6100, 0x6e00,
 	0x6500, 0x6e00, 0x7400, 0x5300, 0x6500, 0x6300,
 	0x7500, 0x7200, 0x6500, 0x6400, 0x0000
 };
-const uint16_t Margin_tool_str[] = {
-	0x4d00,  0x6100,  0x7200,  0x6700,  0x6900,  0x6e00,
-	0x2000,  0x7400,  0x6f00,  0x6f00,  0x6c00,  0x0000,
+/* Gaudi3 Error state sensor 2 */
+const uint16_t _temperature_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7400, 0x6500, 0x6d00, 0x7000, 0x6500, 0x7200,
+	0x6100, 0x7400, 0x7500, 0x7200, 0x6500, 0x0000
 };
-const uint16_t Eth_Serdes_Availability_str[] = {
-	0x4500,  0x7400,  0x6800,  0x2000,  0x5300,  0x6500,
-	0x7200,  0x6400,  0x6500,  0x7300,  0x2000,  0x4100,
-	0x7600,  0x6100,  0x6900,  0x6c00,  0x6100,  0x6200,
-	0x6900,  0x6c00,  0x6900,  0x7400,  0x7900,  0x0000,
+const uint16_t _voltage_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7600, 0x6f00, 0x6c00, 0x7400, 0x6100, 0x6700,
+	0x6500, 0x0000
 };
-const uint16_t asicSerialNumberHigh_str[] = {
-	0x6100, 0x7300, 0x6900, 0x6300, 0x5300, 0x6500, 0x7200,
-	0x6900, 0x6100, 0x6c00, 0x4e00, 0x7500, 0x6d00, 0x6200,
-	0x6500, 0x7200, 0x4800, 0x6900, 0x6700, 0x6800, 0x0000
+const uint16_t _tempRiseTimeErr_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7400, 0x6500, 0x6d00, 0x7000, 0x5200, 0x6900,
+	0x7300, 0x6500, 0x5400, 0x6900, 0x6d00, 0x6500,
+	0x4500, 0x7200, 0x7200, 0x0000
 };
-const uint16_t asicSerialNumberLow_str[] = {
-	0x6100, 0x7300, 0x6900, 0x6300, 0x5300, 0x6500, 0x7200,
-	0x6900, 0x6100, 0x6c00, 0x4e00, 0x7500, 0x6d00, 0x6200,
-	0x6500, 0x7200, 0x4c00, 0x6f00, 0x7700, 0x0000
+const uint16_t _hbmCATTRIP_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6800, 0x6200, 0x6d00, 0x4300, 0x4100, 0x5400,
+	0x5400, 0x5200, 0x4900, 0x5000, 0x0000
 };
-const uint16_t devicePDRRepositoryVersion_str[] = {
-	0x6400, 0x6500, 0x7600, 0x6900, 0x6300, 0x6500, 0x5000,
-	0x4400, 0x5200, 0x5200, 0x6500, 0x7000, 0x6f00, 0x7300,
-	0x6900, 0x7400, 0x6f00, 0x7200, 0x7900, 0x5600, 0x6500,
-	0x7200, 0x7300, 0x6900, 0x6f00, 0x6e00, 0x0000
+const uint16_t _security_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7300, 0x6500, 0x6300, 0x7500, 0x7200, 0x6900,
+	0x7400, 0x7900, 0x0000
 };
-const uint16_t currentPCIeLinkWidth_str[] = {
-	0x6300, 0x7500, 0x7200, 0x7200, 0x6500, 0x6E00, 0x7400, 0x5000,
-	0x4300, 0x4900, 0x6500, 0x4C00, 0x6900, 0x6E00, 0x6B00, 0x5700,
-	0x6900, 0x6400, 0x7400, 0x6800, 0x0000
+const uint16_t _wd_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7700, 0x6400, 0x0000
 };
-const uint16_t currentPCIeLinkSpeed_str[] = {
-	0x6300, 0x7500, 0x7200, 0x7200, 0x6500, 0x6E00, 0x7400, 0x5000,
-	0x4300, 0x4900, 0x6500, 0x4C00, 0x6900, 0x6E00, 0x6B00, 0x5300,
-	0x7000, 0x6500, 0x6500, 0x6400, 0x0000
+const uint16_t _checksum_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6300, 0x6800, 0x6500, 0x6300, 0x6b00, 0x7300,
+	0x7500, 0x6d00, 0x0000
 };
-const uint16_t pcieFWVersion_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4600, 0x5700, 0x5600, 0x6500,
-	0x7200, 0x7300, 0x6900, 0x6f00, 0x6e00, 0x0000
+/* Gaudi3 security SPI error state sensor */
+const uint16_t _revokedPublicKey_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7600, 0x6f00, 0x6b00, 0x6500,
+	0x6400, 0x5000, 0x7500, 0x6200, 0x6c00, 0x6900,
+	0x6300, 0x4b00, 0x6500, 0x7900, 0x0000
 };
-const uint16_t pcieMarginToolModePort0_str[] = {
+const uint16_t _svn_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7300, 0x7600, 0x6e00, 0x0000
+};
+const uint16_t _signature_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7300, 0x6900, 0x6700, 0x6e00, 0x6100, 0x7400,
+	0x7500, 0x7200, 0x6500, 0x0000
+};
+/* Gaudi3 reset effecter */
+const uint16_t _asicReset_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6100, 0x7300, 0x6900, 0x6300, 0x5200, 0x6500,
+	0x7300, 0x6500, 0x7400, 0x0000
+};
+/* Gaudi3 margin tool effecter */
+const uint16_t _marginTool_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6d00, 0x6100, 0x7200, 0x6700, 0x6900, 0x6e00,
+	0x5400, 0x6f00, 0x6f00, 0x6c00, 0x0000
+};
+/* Gaudi3 margin tool SERDES OAM state sensor */
+const uint16_t _oamResult1_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x5200, 0x6500, 0x7300,
+	0x7500, 0x6c00, 0x7400, 0x3100, 0x0000
+};
+const uint16_t _oamResult2_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x5200, 0x6500, 0x7300,
+	0x7500, 0x6c00, 0x7400, 0x3200, 0x0000
+};
+const uint16_t _oamResult3_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x5200, 0x6500, 0x7300,
+	0x7500, 0x6c00, 0x7400, 0x3300, 0x0000
+};
+const uint16_t _oamResult4_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x5200, 0x6500, 0x7300,
+	0x7500, 0x6c00, 0x7400, 0x3400, 0x0000
+};
+const uint16_t _oamResult5_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x5200, 0x6500, 0x7300,
+	0x7500, 0x6c00, 0x7400, 0x3500, 0x0000
+};
+const uint16_t _oamResult6_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x5200, 0x6500, 0x7300,
+	0x7500, 0x6c00, 0x7400, 0x3600, 0x0000
+};
+const uint16_t _oamResult7_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x5200, 0x6500, 0x7300,
+	0x7500, 0x6c00, 0x7400, 0x3700, 0x0000
+};
+const uint16_t _oamResult8_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x5200, 0x6500, 0x7300,
+	0x7500, 0x6c00, 0x7400, 0x3800, 0x0000
+};
+/* Gaudi3 margin tool SERDES retimer state sensor 1 */
+const uint16_t _retimer1Result1_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3100, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3100, 0x0000
+};
+const uint16_t _retimer1Result2_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3100, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3200, 0x0000
+};
+const uint16_t _retimer1Result3_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3100, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3300, 0x0000
+};
+const uint16_t _retimer1Result4_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3100, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3400, 0x0000
+};
+const uint16_t _retimer1Result5_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3100, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3500, 0x0000
+};
+const uint16_t _retimer1Result6_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3100, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3600, 0x0000
+};
+/* Gaudi3 margin tool SERDES retimer state sensor 2 */
+const uint16_t _retimer2Result1_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3200, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3100, 0x0000
+};
+const uint16_t _retimer2Result2_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3200, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3200, 0x0000
+};
+const uint16_t _retimer2Result3_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3200, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3300, 0x0000
+};
+const uint16_t _retimer2Result4_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7200, 0x6500, 0x7400, 0x6900, 0x6d00, 0x6500,
+	0x7200, 0x3200, 0x5200, 0x6500, 0x7300, 0x7500,
+	0x6c00, 0x7400, 0x3400, 0x0000
+};
+/* Gaudi3 margin tool PCIe lanes state effecter 1 */
+const uint16_t _pcieMarginToolModePort0_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3000, 0x0000
 };
-const uint16_t pcieMarginToolModePort1_str[] = {
+const uint16_t _pcieMarginToolModePort1_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x0000
 };
-const uint16_t pcieMarginToolModePort2_str[] = {
+const uint16_t _pcieMarginToolModePort2_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3200, 0x0000
 };
-const uint16_t pcieMarginToolModePort3_str[] = {
+const uint16_t _pcieMarginToolModePort3_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3300, 0x0000
 };
-const uint16_t pcieMarginToolModePort4_str[] = {
+const uint16_t _pcieMarginToolModePort4_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3400, 0x0000
 };
-const uint16_t pcieMarginToolModePort5_str[] = {
+const uint16_t _pcieMarginToolModePort5_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3500, 0x0000
 };
-const uint16_t pcieMarginToolModePort6_str[] = {
+const uint16_t _pcieMarginToolModePort6_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3600, 0x0000
 };
-const uint16_t pcieMarginToolModePort7_str[] = {
+const uint16_t _pcieMarginToolModePort7_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3700, 0x0000
 };
-const uint16_t pcieMarginToolModePort8_str[] = {
+/* Gaudi3 margin tool PCIe lanes state effecter 2 */
+const uint16_t _pcieMarginToolModePort8_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3800, 0x0000
 };
-const uint16_t pcieMarginToolModePort9_str[] = {
+const uint16_t _pcieMarginToolModePort9_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3900, 0x0000
 };
-const uint16_t pcieMarginToolModePort10_str[] = {
+const uint16_t _pcieMarginToolModePort10_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3000,
 	0x0000
 };
-const uint16_t pcieMarginToolModePort11_str[] = {
+const uint16_t _pcieMarginToolModePort11_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3100,
 	0x0000
 };
-const uint16_t pcieMarginToolModePort12_str[] = {
+const uint16_t _pcieMarginToolModePort12_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3200,
 	0x0000
 };
-const uint16_t pcieMarginToolModePort13_str[] = {
+const uint16_t _pcieMarginToolModePort13_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3300,
 	0x0000
 };
-const uint16_t pcieMarginToolModePort14_str[] = {
+const uint16_t _pcieMarginToolModePort14_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3400,
 	0x0000
 };
-const uint16_t pcieMarginToolModePort15_str[] = {
+const uint16_t _pcieMarginToolModePort15_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x7000, 0x6300, 0x6900, 0x6500, 0x4d00, 0x6100,
 	0x7200, 0x6700, 0x6900, 0x6e00, 0x5400, 0x6f00,
 	0x6f00, 0x6c00, 0x4d00, 0x6f00, 0x6400, 0x6500,
 	0x5000, 0x6f00, 0x7200, 0x7400, 0x3100, 0x3500,
 	0x0000
 };
-const uint16_t pcieDeviceID_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x4400, 0x6500, 0x7600,
-	0x6900, 0x6300, 0x6500, 0x4900, 0x4400, 0x0000
+/* Gaudi3 identification PCIe device ID */
+const uint16_t _pcieDeviceID_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x4400, 0x6500,
+	0x7600, 0x6900, 0x6300, 0x6500, 0x4900, 0x4400,
+	0x0000
 };
-const uint16_t pcieVendorID_str[] = {
-	0x7000, 0x6300, 0x6900, 0x6500, 0x5600, 0x6500, 0x6e00,
-	0x6400, 0x6f00, 0x7200, 0x4900, 0x4400, 0x0000
+/* Gaudi3 identification PCIe vendor ID */
+const uint16_t _pcieVendorID_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x7000, 0x6300, 0x6900, 0x6500, 0x5600, 0x6500,
+	0x6e00, 0x6400, 0x6f00, 0x7200, 0x4900, 0x4400,
+	0x0000
 };
 
-const uint16_t In_preboot_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x7000, 0x7200, 0x6500, 0x6200,
-	0x6F00, 0x6F00, 0x7400, 0x0000
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+
+struct pldm_pdr_name_record pldm_pdr_name_records[] OFFLOAD_PDR_ATTR = {
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_ASIC_SERIAL_NUM_HIGH,
+			.count = 1,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_asicSerialNumberHigh_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _asicSerialNumberHigh_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_ASIC_SERIAL_NUM_LOW,
+			.count = 1,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_asicSerialNumberLow_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _asicSerialNumberLow_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_DEVICE_PDR_REPOSITORY_VERSION,
+			.count = 1,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_devicePDRRepositoryVersion_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _devicePDRRepositoryVersion_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_PCIE_LINK_WIDTH,
+			.count = 1,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_currentPCIeLinkWidth_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _currentPCIeLinkWidth_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_SPEED] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_PCIE_LINK_SPEED,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_PCIE_LINK_SPEED,
+			.count = 1,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_currentPCIeLinkSpeed_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _currentPCIeLinkSpeed_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_FW_STATE] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_FW_STATE,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_FW_STATE,
+			.count = 2,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_bootStage_str,
+				_fwUpdateState_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _bootStage_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _fwUpdateState_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_KEY_STATE] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_SECURITY_KEY_STATE,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_SECURITY_KEY_STATE,
+			.count = 5,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_key1Revocation_str,
+				_key2Revocation_str,
+				_key3Revocation_str,
+				_key4Revocation_str,
+				_key5Revocation_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _key1Revocation_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _key2Revocation_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _key3Revocation_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _key4Revocation_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _key5Revocation_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_STATE] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_SECURITY_STATE,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_SECURITY_STATE,
+			.count = 4,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_fwImageSource_str,
+				_memoryCleanup_str,
+				_spiWriteProtect_str,
+				_permanentSecured_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _fwImageSource_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _memoryCleanup_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _spiWriteProtect_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _permanentSecured_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ERROR_STATE_2] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_ERROR_STATE_2,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_ERROR_STATE_2,
+			.count = 7,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_temperature_str,
+				_voltage_str,
+				_tempRiseTimeErr_str,
+				_hbmCATTRIP_str,
+				_security_str,
+				_wd_str,
+				_checksum_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _temperature_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _voltage_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _tempRiseTimeErr_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _hbmCATTRIP_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _security_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _wd_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _checksum_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_SECURITY_SPI_ERROR_STATE,
+			.count = 3,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_revokedPublicKey_str,
+				_svn_str,
+				_signature_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _revokedPublicKey_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _svn_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _signature_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_EFFECTER_AUX_NAMES_ASIC_RESET] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_EFFECTER_AUX_NAMES_ASIC_RESET,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_ASIC_RESET,
+			.count = 1,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_asicReset_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _asicReset_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_EFFECTER_AUX_NAMES_MARGIN_TOOL] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_EFFECTER_AUX_NAMES_MARGIN_TOOL,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_MARGIN_TOOL,
+			.count = 1,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_marginTool_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _marginTool_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_ETH_SERDES_AVAILABILITY_OAM,
+			.count = 8,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_oamResult1_str,
+				_oamResult2_str,
+				_oamResult3_str,
+				_oamResult4_str,
+				_oamResult5_str,
+				_oamResult6_str,
+				_oamResult7_str,
+				_oamResult8_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamResult1_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamResult2_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamResult3_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamResult4_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamResult5_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamResult6_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamResult7_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _oamResult8_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_ETH_SERDES_AVAILABILITY_RETIMER_1,
+			.count = 6,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_retimer1Result1_str,
+				_retimer1Result2_str,
+				_retimer1Result3_str,
+				_retimer1Result4_str,
+				_retimer1Result5_str,
+				_retimer1Result6_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer1Result1_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer1Result2_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer1Result3_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer1Result4_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer1Result5_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer1Result6_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_ETH_SERDES_AVAILABILITY_RETIMER_2,
+			.count = 4,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_retimer2Result1_str,
+				_retimer2Result2_str,
+				_retimer2Result3_str,
+				_retimer2Result4_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer2Result1_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer2Result2_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer2Result3_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _retimer2Result4_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_PCIE_MARGIN_TOOL_MODE_1,
+			.count = 8,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_pcieMarginToolModePort0_str,
+				_pcieMarginToolModePort1_str,
+				_pcieMarginToolModePort2_str,
+				_pcieMarginToolModePort3_str,
+				_pcieMarginToolModePort4_str,
+				_pcieMarginToolModePort5_str,
+				_pcieMarginToolModePort6_str,
+				_pcieMarginToolModePort7_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort0_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort1_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort2_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort3_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort4_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort5_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort6_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort7_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_PCIE_MARGIN_TOOL_MODE_2,
+			.count = 8,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_pcieMarginToolModePort8_str,
+				_pcieMarginToolModePort9_str,
+				_pcieMarginToolModePort10_str,
+				_pcieMarginToolModePort11_str,
+				_pcieMarginToolModePort12_str,
+				_pcieMarginToolModePort13_str,
+				_pcieMarginToolModePort14_str,
+				_pcieMarginToolModePort15_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort8_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort9_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort10_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort11_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort12_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort13_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort14_str,
+				},
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieMarginToolModePort15_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_DEVICE_ID] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_PCIE_DEVICE_ID,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_PCIE_DEVICE_ID,
+			.count = 1,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_pcieDeviceID_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieDeviceID_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_VENDOR_ID] = {
+		.aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_AUX_NAMES_PCIE_VENDOR_ID,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id = PLDM_SID_PCIE_VENDOR_ID,
+			.count = 1,
+			.names = {
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+				_pcieVendorID_str,
+#else /* old aux name format */
+				{
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+					.name = _pcieVendorID_str,
+				},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+			},
+		},
+	},
 };
-const uint16_t In_management_FW_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x6D00, 0x6100, 0x6E00, 0x6100,
-	0x6700, 0x6500, 0x6D00, 0x6500, 0x6E00, 0x7400, 0x2000,
-	0x4600, 0x5700, 0x0000
+const size_t pldm_pdr_name_records_length = ARRAY_SIZE(pldm_pdr_name_records);
+
+/**** Auxilary Name PDRs ****/
+
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+
+/* Gaudi3 OAM add-in card accelerator */
+#define _oamAddInCardAccelerator_str "oamAddInCardAccelerator"
+/* Gaudi3 OAM CPLD */
+#define _oamCPLD_str "oamCPLD"
+/* Gaudi3 accelerator */
+#define _accelerator_str "accelerator"
+
+#else /* old aux name format */
+
+/* Gaudi3 OAM add-in card accelerator */
+const uint16_t _oamAddInCardAccelerator_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x4100, 0x6400, 0x6400,
+	0x4900, 0x6e00, 0x4300, 0x6100, 0x7200, 0x6400,
+	0x4100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500,
+	0x7200, 0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
 };
-const uint16_t In_preboot_recovery_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x7000, 0x7200, 0x6500, 0x6200,
-	0x6F00, 0x6F00, 0x7400, 0x2000, 0x7200, 0x6500, 0x6300,
-	0x6F00, 0x7600, 0x6500, 0x7200, 0x7900, 0x0000
+/* Gaudi3 OAM CPLD */
+const uint16_t _oamCPLD_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6f00, 0x6100, 0x6d00, 0x4300, 0x5000, 0x4c00,
+	0x4400, 0x0000
 };
-const uint16_t In_margin_tool_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x6D00, 0x6100, 0x7200, 0x6700,
-	0x6900, 0x6E00, 0x2000, 0x7400, 0x6F00, 0x6F00, 0x6C00,
+/* Gaudi3 accelerator */
+const uint16_t _accelerator_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x6100, 0x6300, 0x6300, 0x6500, 0x6c00, 0x6500,
+	0x7200, 0x6100, 0x7400, 0x6f00, 0x7200, 0x0000
+};
+
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+
+struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[] OFFLOAD_PDR_ATTR = {
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.shared_name_count = 0,
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+			.name = _oamAddInCardAccelerator_str,
+#else /* old aux name format */
+			.name = {
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = _oamAddInCardAccelerator_str,
+			},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+		},
+	},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_CPLD,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.shared_name_count = 0,
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+			.name = _oamCPLD_str,
+#else /* old aux name format */
+			.name = {
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = _oamCPLD_str,
+			},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+		},
+	},
+	[PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR] = {
+		.entity_aux_names = {
+			.header = {
+				.record_handle =
+					PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
+				.record_change_number = 0,
+			},
+			.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
+			.entity_instance = 1,
+			.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
+			.shared_name_count = 0,
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+			.name = _accelerator_str,
+#else /* old aux name format */
+			.name = {
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = _accelerator_str,
+			},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+		},
+	},
+};
+const size_t pldm_pdr_entity_name_records_length = ARRAY_SIZE(pldm_pdr_entity_name_records);
+
+/**** OEM State Set PDRs ****/
+
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+
+/* FW state set */
+#define _In_preBoot_str "In preBoot"
+#define _In_management_FW_str "In management FW"
+#define _In_preBoot_recovery_str "In preBoot recovery"
+#define _In_Margin_tool_str "In Margin tool"
+#define _In_FW_loader_agent_str "In FW loader agent"
+/* FW ppBoot source */
+#define _Primary_str "Primary"
+#define _Secondary_str "Secondary"
+/* Security key revocation */
+#define _Not_revoked_str "Not revoked"
+#define _Revoked_str "Revoked"
+/* Security mode status */
+#define _Not_permanent_str "Not permanent"
+#define _Permanent_str "Permanent"
+/* SPI flash W/P state */
+#define _Not_asserted_str "Not asserted"
+#define _Asserted_str "Asserted"
+/* PCIe margin tool mode */
+#define _Only_BER_str "Only BER"
+#define _Only_diagnostics_str "Only diagnostics"
+#define _BER_and_Diagnostics_str "BER and Diagnostics"
+#define _Skip_str "Skip"
+
+#else /* old aux name format */
+
+/* FW state set */
+const uint16_t _In_preBoot_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4900, 0x6e00, 0x2000, 0x7000, 0x7200, 0x6500,
+	0x4200, 0x6f00, 0x6f00, 0x7400, 0x0000
+};
+const uint16_t _In_management_FW_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4900, 0x6e00, 0x2000, 0x6d00, 0x6100, 0x6e00,
+	0x6100, 0x6700, 0x6500, 0x6d00, 0x6500, 0x6e00,
+	0x7400, 0x2000, 0x4600, 0x5700, 0x0000
+};
+const uint16_t _In_preBoot_recovery_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4900, 0x6e00, 0x2000, 0x7000, 0x7200, 0x6500,
+	0x4200, 0x6f00, 0x6f00, 0x7400, 0x2000, 0x7200,
+	0x6500, 0x6300, 0x6f00, 0x7600, 0x6500, 0x7200,
+	0x7900, 0x0000
+};
+const uint16_t _In_Margin_tool_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4900, 0x6e00, 0x2000, 0x4d00, 0x6100, 0x7200,
+	0x6700, 0x6900, 0x6e00, 0x2000, 0x7400, 0x6f00,
+	0x6f00, 0x6c00, 0x0000
+};
+const uint16_t _In_FW_loader_agent_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4900, 0x6e00, 0x2000, 0x4600, 0x5700, 0x2000,
+	0x6c00, 0x6f00, 0x6100, 0x6400, 0x6500, 0x7200,
+	0x2000, 0x6100, 0x6700, 0x6500, 0x6e00, 0x7400,
 	0x0000
 };
-const uint16_t In_FW_loader_agent_str[] = {
-	0x4900, 0x6E00, 0x2000, 0x4600, 0x5700, 0x2000, 0x6C00,
-	0x6F00, 0x6100, 0x6400, 0x6500, 0x7200, 0x2000, 0x6100,
-	0x6700, 0x6500, 0x6E00, 0x7400, 0x0000
+/* FW ppBoot source */
+const uint16_t _Primary_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x5000, 0x7200, 0x6900, 0x6d00, 0x6100, 0x7200,
+	0x7900, 0x0000
 };
-const uint16_t Primary_str[] = {
-	0x5000, 0x7200, 0x6900, 0x6D00, 0x6100, 0x7200, 0x7900,
-	0x0000
+const uint16_t _Secondary_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x5300, 0x6500, 0x6300, 0x6f00, 0x6e00, 0x6400,
+	0x6100, 0x7200, 0x7900, 0x0000
 };
-const uint16_t Secondary_str[] = {
-	0x5300, 0x6500, 0x6300, 0x6F00, 0x6E00, 0x6400, 0x6100,
-	0x7200, 0x7900, 0x0000
+/* Security key revocation */
+const uint16_t _Not_revoked_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4e00, 0x6f00, 0x7400, 0x2000, 0x7200, 0x6500,
+	0x7600, 0x6f00, 0x6b00, 0x6500, 0x6400, 0x0000
 };
-const uint16_t Not_revoked_str[] = {
-	0x4E00, 0x6F00, 0x7400, 0x2000, 0x7200, 0x6500, 0x7600,
-	0x6F00, 0x6B00, 0x6500, 0x6400, 0x0000
-};
-const uint16_t Revoked_str[] = {
-	0x5200, 0x6500, 0x7600, 0x6F00, 0x6B00, 0x6500, 0x6400,
-	0x0000
-};
-const uint16_t Not_permanent_str[] = {
-	0x4E00, 0x6F00, 0x7400, 0x2000, 0x7000, 0x6500, 0x7200,
-	0x6D00, 0x6100, 0x6E00, 0x6500, 0x6E00, 0x7400, 0x0000
-};
-const uint16_t Permanent_str[] = {
-	0x5000, 0x6500, 0x7200, 0x6D00, 0x6100, 0x6E00, 0x6500,
-	0x6E00, 0x7400, 0x0000
-};
-const uint16_t Not_asserted_str[] = {
-	0x4E00, 0x6F00, 0x7400, 0x2000, 0x6100, 0x7300, 0x7300,
-	0x6500, 0x7200, 0x7400, 0x6500, 0x6400, 0x0000
-};
-const uint16_t Asserted_str[] = {
-	0x4100, 0x7300, 0x7300, 0x6500, 0x7200, 0x7400, 0x6500,
+const uint16_t _Revoked_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x5200, 0x6500, 0x7600, 0x6f00, 0x6b00, 0x6500,
 	0x6400, 0x0000
 };
-const uint16_t Only_ber_str[] = {
-	0x4F00, 0x6E00, 0x6C00, 0x7900, 0x2000, 0x4200, 0x4500,
-	0x5200, 0x0000
+/* Security mode status */
+const uint16_t _Not_permanent_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4e00, 0x6f00, 0x7400, 0x2000, 0x7000, 0x6500,
+	0x7200, 0x6d00, 0x6100, 0x6e00, 0x6500, 0x6e00,
+	0x7400, 0x0000
 };
-const uint16_t Only_diagnostics_str[] = {
-	0x4F00, 0x6E00, 0x6C00, 0x7900, 0x2000, 0x6400, 0x6900,
-	0x6100, 0x6700, 0x6E00, 0x6F00, 0x7300, 0x7400, 0x6900,
-	0x6300, 0x7300, 0x0000
+const uint16_t _Permanent_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x5000, 0x6500, 0x7200, 0x6d00, 0x6100, 0x6e00,
+	0x6500, 0x6e00, 0x7400, 0x0000
 };
-const uint16_t Ber_and_diagnostics_str[] = {
-	0x4200, 0x4500, 0x5200, 0x2000, 0x6100, 0x6E00, 0x6400,
-	0x2000, 0x6400, 0x6900, 0x6100, 0x6700, 0x6E00, 0x6F00,
-	0x7300, 0x7400, 0x6900, 0x6300, 0x7300, 0x0000
+/* SPI flash W/P state */
+const uint16_t _Not_asserted_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4e00, 0x6f00, 0x7400, 0x2000, 0x6100, 0x7300,
+	0x7300, 0x6500, 0x7200, 0x7400, 0x6500, 0x6400,
+	0x0000
 };
-const uint16_t Skip_str[] = {
-	0x5300, 0x6B00, 0x6900, 0x7000, 0x0000
+const uint16_t _Asserted_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4100, 0x7300, 0x7300, 0x6500, 0x7200, 0x7400,
+	0x6500, 0x6400, 0x0000
 };
-const uint16_t Ver32_str[] = {
+/* PCIe margin tool mode */
+const uint16_t _Only_BER_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4f00, 0x6e00, 0x6c00, 0x7900, 0x2000, 0x4200,
+	0x4500, 0x5200, 0x0000
+};
+const uint16_t _Only_diagnostics_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4f00, 0x6e00, 0x6c00, 0x7900, 0x2000, 0x6400,
+	0x6900, 0x6100, 0x6700, 0x6e00, 0x6f00, 0x7300,
+	0x7400, 0x6900, 0x6300, 0x7300, 0x0000
+};
+const uint16_t _BER_and_Diagnostics_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x4200, 0x4500, 0x5200, 0x2000, 0x6100, 0x6e00,
+	0x6400, 0x2000, 0x4400, 0x6900, 0x6100, 0x6700,
+	0x6e00, 0x6f00, 0x7300, 0x7400, 0x6900, 0x6300,
+	0x7300, 0x0000
+};
+const uint16_t _Skip_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x5300, 0x6b00, 0x6900, 0x7000, 0x0000
+};
+
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+
+struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[] OFFLOAD_PDR_ATTR = {
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE] = {
+		.oem_state_set = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_OEM_STATE_SET_FW_STATE,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id_handle = 32769,
+			.iana = INTEL_IANA,
+			.id = 32769,
+			.count = 5,
+			.values = {
+				{
+					.min_value = 1,
+					.max_value = 1,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _In_preBoot_str,
+				},
+				{
+					.min_value = 2,
+					.max_value = 2,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _In_management_FW_str,
+				},
+				{
+					.min_value = 3,
+					.max_value = 3,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _In_preBoot_recovery_str,
+				},
+				{
+					.min_value = 4,
+					.max_value = 4,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _In_Margin_tool_str,
+				},
+				{
+					.min_value = 5,
+					.max_value = 5,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _In_FW_loader_agent_str,
+				},
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE] = {
+		.oem_state_set = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id_handle = 32770,
+			.iana = INTEL_IANA,
+			.id = 32770,
+			.count = 2,
+			.values = {
+				{
+					.min_value = 1,
+					.max_value = 1,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Primary_str,
+				},
+				{
+					.min_value = 2,
+					.max_value = 2,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Secondary_str,
+				},
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_KEY] = {
+		.oem_state_set = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_OEM_STATE_SET_SECURITY_KEY,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id_handle = 32771,
+			.iana = INTEL_IANA,
+			.id = 32771,
+			.count = 2,
+			.values = {
+				{
+					.min_value = 1,
+					.max_value = 1,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Not_revoked_str,
+				},
+				{
+					.min_value = 2,
+					.max_value = 2,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Revoked_str,
+				},
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS] = {
+		.oem_state_set = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id_handle = 32783,
+			.iana = INTEL_IANA,
+			.id = 32783,
+			.count = 2,
+			.values = {
+				{
+					.min_value = 1,
+					.max_value = 1,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Not_permanent_str,
+				},
+				{
+					.min_value = 2,
+					.max_value = 2,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Permanent_str,
+				},
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE] = {
+		.oem_state_set = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id_handle = 32784,
+			.iana = INTEL_IANA,
+			.id = 32784,
+			.count = 2,
+			.values = {
+				{
+					.min_value = 1,
+					.max_value = 1,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Not_asserted_str,
+				},
+				{
+					.min_value = 2,
+					.max_value = 2,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Asserted_str,
+				},
+			},
+		},
+	},
+	[PLDM_RIDX_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE] = {
+		.oem_state_set = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_OEM_STATE_SET,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.id_handle = 32785,
+			.iana = INTEL_IANA,
+			.id = 32785,
+			.count = 4,
+			.values = {
+				{
+					.min_value = 1,
+					.max_value = 1,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Only_BER_str,
+				},
+				{
+					.min_value = 2,
+					.max_value = 2,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Only_diagnostics_str,
+				},
+				{
+					.min_value = 3,
+					.max_value = 3,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _BER_and_Diagnostics_str,
+				},
+				{
+					.min_value = 4,
+					.max_value = 4,
+#if !CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+					.name_string_count = 1,
+					.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
+					.name = _Skip_str,
+				},
+			},
+		},
+	},
+};
+const size_t pldm_pdr_oem_state_set_records_length = ARRAY_SIZE(pldm_pdr_oem_state_set_records);
+
+/**** OEM Unit PDRs ****/
+
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+
+/* Ver32 */
+#define _Ver32_str "Ver32"
+/* String32 */
+#define _String32_str "String32"
+/* Hex16 */
+#define _Hex16_str "Hex16"
+
+#else /* old aux name format */
+
+/* Ver32 */
+const uint16_t _Ver32_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x5600, 0x6500, 0x7200, 0x3300, 0x3200, 0x0000
 };
-const uint16_t String32_str[] = {
-	0x5300, 0x7400, 0x7200, 0x6900, 0x6e00, 0x6700, 0x3300,
-	0x3200, 0x0000
+/* String32 */
+const uint16_t _String32_str[] OFFLOAD_AUX_NAMES_ATTR = {
+	0x5300, 0x7400, 0x7200, 0x6900, 0x6e00, 0x6700,
+	0x3300, 0x3200, 0x0000
 };
-const uint16_t Hex16_str[] = {
+/* Hex16 */
+const uint16_t _Hex16_str[] OFFLOAD_AUX_NAMES_ATTR = {
 	0x4800, 0x6500, 0x7800, 0x3100, 0x3600, 0x0000
 };
 
-struct pldm_pdr_name_record pldm_pdr_name_records[] = {
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_HIGH,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ASIC_SERIAL_NUM_HIGH,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = asicSerialNumberHigh_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ASIC_SERIAL_NUM_LOW,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ASIC_SERIAL_NUM_LOW,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = asicSerialNumberLow_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_DEVICE_PDR_REPOSITORY_VERSION,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_DEVICE_PDR_REPOSITORY_VERSION,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = devicePDRRepositoryVersion_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_LINK_WIDTH,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_LINK_WIDTH,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = currentPCIeLinkWidth_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_LINK_SPEED] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_LINK_SPEED,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_LINK_SPEED,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = currentPCIeLinkSpeed_str,
-		}},
-	}},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
 
-
-	[PLDM_RIDX_SENSOR_AUX_NAMES_FW_STATE] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_FW_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
+struct pldm_pdr_oem_unit_record pldm_pdr_oem_unit_records[] OFFLOAD_PDR_ATTR = {
+	[PLDM_RIDX_SENSOR_OEM_UNIT_VER_32] = {
+		.oem_unit = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_OEM_UNIT_VER_32,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_OEM_UNIT,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.oem_unit_handle = 0,
+			.iana = INTEL_IANA,
+			.oem_unit_id = 0,
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+			.name = _Ver32_str,
+#else
+			.name = {
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = _Ver32_str,
+			},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
 		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_FW_STATE,
-		.count = 2,
-		.names = {
-			{
+	},
+	[PLDM_RIDX_SENSOR_OEM_UNIT_STRING_32] = {
+		.oem_unit = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_OEM_UNIT_STRING_32,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_OEM_UNIT,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.oem_unit_handle = 2,
+			.iana = INTEL_IANA,
+			.oem_unit_id = 2,
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+			.name = _String32_str,
+#else
+			.name = {
 				.name_string_count = 1,
 				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = bootStage_str,
+				.name = _String32_str,
 			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = fwUpdateState_str,
-			},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
 		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ERROR_STATE_2] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ERROR_STATE_2,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
+	},
+	[PLDM_RIDX_SENSOR_OEM_UNIT_HEX_16] = {
+		.oem_unit = {
+			.header = {
+				.record_handle =
+					PLDM_RH_SENSOR_OEM_UNIT_HEX_16,
+				.header_version = PLDM_PDR_HEADER_VERSION,
+				.type = PLDM_PDR_TYPE_OEM_UNIT,
+				.record_change_number = 0,
+			},
+			.terminus_handle = 0,
+			.oem_unit_handle = 3,
+			.iana = INTEL_IANA,
+			.oem_unit_id = 3,
+#if CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES
+			.name = _Hex16_str,
+#else
+			.name = {
+				.name_string_count = 1,
+				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
+				.name = _Hex16_str,
+			},
+#endif /* CONFIG_LIBPLDM_PDR_ASCII_AUX_NAMES */
 		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ERROR_STATE_2,
-		.count = 7,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = temperature_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = voltage_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = tempRiseTimeErr_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = hbmCATTRIP_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = security_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = wd_str,
-			},			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = checksum_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_SECURITY_SPI_ERROR_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_SECURITY_SPI_ERROR_STATE,
-		.count = 3,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = revokedPublicKey_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = svn_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = signature_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_EFFECTER_AUX_NAMES_ASIC_RESET] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_ASIC_RESET,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ASIC_RESET,
-		.count = 1,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = asicReset_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_KEY_STATE] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_SECURITY_KEY_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_SECURITY_KEY_STATE,
-		.count = 5,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key1Revocation_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key2Revocation_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key3Revocation_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key4Revocation_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = key5Revocation_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_SECURITY_STATE] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_SECURITY_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_SECURITY_STATE,
-		.count = 4,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = fwImageSource_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = memoryCleanup_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = spiWriteProtect_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = permanentSecured_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_EFFECTER_AUX_NAMES_MARGIN_TOOL] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_MARGIN_TOOL,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_MARGIN_TOOL,
-		.count = 1,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Margin_tool_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_OAM,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ETH_SERDES_AVAILABILITY_OAM,
-		.count = 8,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_1,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ETH_SERDES_AVAILABILITY_RETIMER_1,
-		.count = 6,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_ETH_SERDES_AVAILABILITY_RETIMER_2,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_ETH_SERDES_AVAILABILITY_RETIMER_2,
-		.count = 4,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Eth_Serdes_Availability_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_1,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_MARGIN_TOOL_MODE_1,
-		.count = 8,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort0_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort1_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort2_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort3_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort4_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort5_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort6_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort7_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_EFFECTER_AUX_NAMES_PCIE_MARGIN_TOOL_MODE_2,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_EFFECTER_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_MARGIN_TOOL_MODE_2,
-		.count = 8,
-		.names = {
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort8_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort9_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort10_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort11_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort12_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort13_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort14_str,
-			},
-			{
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = pcieMarginToolModePort15_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_DEVICE_ID] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_DEVICE_ID,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_DEVICE_ID,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name =  pcieDeviceID_str,
-		}},
-	}},
-	[PLDM_RIDX_SENSOR_AUX_NAMES_PCIE_VENDOR_ID] = { .aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_AUX_NAMES_PCIE_VENDOR_ID,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_SENSOR_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id = PLDM_SID_PCIE_VENDOR_ID,
-		.count = 1,
-		.names = {{
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = pcieVendorID_str,
-		}},
-	}},
+	},
 };
-
-const size_t pldm_pdr_name_records_length = ARRAY_SIZE(pldm_pdr_name_records);
-
-struct pldm_pdr_entity_name_record pldm_pdr_entity_name_records[] = {
-	[PLDM_RIDX_ENTITY_AUX_NAMES_ADDIN_CARD] = { .entity_aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ADDIN_CARD,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.entity_type = PLDM_ENTITY_TYPE_ADDIN_CARD,
-		.entity_instance = 1,
-		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.shared_name_count = 0,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = oamAddInCardAccelerator_str,
-		},
-	}},
-	[PLDM_RIDX_ENTITY_AUX_NAMES_OAM_CPLD] = { .entity_aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_OAM_CPLD,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.entity_type = PLDM_ENTITY_TYPE_CPLD,
-		.entity_instance = 1,
-		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.shared_name_count = 0,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = oamCPLD_str,
-		},
-	}},
-	[PLDM_RIDX_ENTITY_AUX_NAMES_ACCELERATOR] = { .entity_aux_names = {
-		.header = {
-			.record_handle = PLDM_RH_ENTITY_AUX_NAMES_ACCELERATOR,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_ENTITY_AUX_NAMES,
-			.record_change_number = 0,
-		},
-		.entity_type = PLDM_ENTITY_TYPE_ACCELERATOR,
-		.entity_instance = 1,
-		.entity_container_id = PLDM_CONTAINER_ADDIN_CARD,
-		.shared_name_count = 0,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = accelerator_str,
-		},
-	}},
-};
-
-const size_t pldm_pdr_entity_name_records_length = ARRAY_SIZE(pldm_pdr_entity_name_records);
-
-struct pldm_pdr_oem_state_set_record pldm_pdr_oem_state_set_records[] = {
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_FW_STATE] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_FW_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_FW_STATE,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_FW_STATE,
-		.count = 5,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_preboot_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_management_FW_str,
-			},
-			{
-				.min_value = 3,
-				.max_value = 3,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_preboot_recovery_str,
-			},
-			{
-				.min_value = 4,
-				.max_value = 4,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_margin_tool_str,
-			},
-			{
-				.min_value = 5,
-				.max_value = 5,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = In_FW_loader_agent_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_PPBOOT_IMAGE_SOURCE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_PPBOOT_IMAGE_SOURCE,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_PPBOOT_IMAGE_SOURCE,
-		.count = 2,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Primary_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Secondary_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_KEY] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_SECURITY_KEY,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_SECURITY_KEY,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_SECURITY_KEY,
-		.count = 2,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Not_revoked_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Revoked_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_SECURITY_MODE_STATUS,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_SECURITY_MODE_STATUS,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_SECURITY_MODE_STATUS,
-		.count = 2,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Not_permanent_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Permanent_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_SPI_FLASH_WRITE_PROTECT_STATE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_SPI_FLASH_WRITE_PROTECT_STATE,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_SPI_FLASH_WRITE_PROTECT_STATE,
-		.count = 2,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Not_asserted_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Asserted_str,
-			},
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE] = { .oem_state_set = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_STATE_SET_PCIE_MARGIN_TOOL_MODE,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_STATE_SET,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.id_handle = PLDM_STATE_SET_OEM_PCIE_MARGIN_TOOL_MODE,
-		.iana = INTEL_IANA,
-		.id = PLDM_STATE_SET_OEM_PCIE_MARGIN_TOOL_MODE,
-		.count = 4,
-		.values = {
-			{
-				.min_value = 1,
-				.max_value = 1,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Only_ber_str,
-			},
-			{
-				.min_value = 2,
-				.max_value = 2,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Only_diagnostics_str,
-			},
-			{
-				.min_value = 3,
-				.max_value = 3,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Ber_and_diagnostics_str,
-			},
-			{
-				.min_value = 4,
-				.max_value = 4,
-				.name_string_count = 1,
-				.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-				.name = Skip_str,
-			},
-		},
-	}},
-};
-
-const size_t pldm_pdr_oem_state_set_records_length = ARRAY_SIZE(pldm_pdr_oem_state_set_records);
-
-struct pldm_pdr_oem_unit_record pldm_pdr_oem_unit_records[] = {
-	[PLDM_RIDX_SENSOR_OEM_UNIT_VER_32] = { .oem_unit = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_UNIT_VER_32,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_UNIT,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.oem_unit_handle = PLDM_BASE_OEM_UNIT_HANDLE_VER_32,
-		.iana = INTEL_IANA,
-		.oem_unit_id = PLDM_BASE_OEM_UNIT_HANDLE_VER_32,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = Ver32_str,
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_UNIT_STRING_32] = { .oem_unit = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_UNIT_STRING_32,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_UNIT,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.oem_unit_handle = PLDM_BASE_OEM_UNIT_HANDLE_STRING_32,
-		.iana = INTEL_IANA,
-		.oem_unit_id = PLDM_BASE_OEM_UNIT_HANDLE_STRING_32,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = String32_str,
-		},
-	}},
-	[PLDM_RIDX_SENSOR_OEM_UNIT_HEX_16] = { .oem_unit = {
-		.header = {
-			.record_handle = PLDM_RH_SENSOR_OEM_UNIT_HEX_16,
-			.header_version = PLDM_PDR_HEADER_VERSION,
-			.type = PLDM_PDR_TYPE_OEM_UNIT,
-			.record_change_number = 0,
-		},
-		.terminus_handle = 0,
-		.oem_unit_handle = PLDM_BASE_OEM_UNIT_HANDLE_HEX_16,
-		.iana = INTEL_IANA,
-		.oem_unit_id = PLDM_BASE_OEM_UNIT_HANDLE_HEX_16,
-		.name = {
-			.name_string_count = 1,
-			.name_language_tag = PLDM_PDR_LANGUAGE_TAG_ENGLISH,
-			.name = Hex16_str,
-		},
-	}},
-};
-
 const size_t pldm_pdr_oem_unit_records_length = ARRAY_SIZE(pldm_pdr_oem_unit_records);
 
